src/net/sourceforge/kolmafia/request/PlaceRequest.java

```diff
@@ -339,6 +344,31 @@ else if (responseText.contains("a giant jagged hole in the wall")) {
     SpadingManager.processPlace(urlString, responseText);
   }
 
+  private static void parseSotVisit(String responseText) {
+    String location = "";
+    // What he means is that he lost something over in <location> and he'd like you to get it for
+    // him.
+    Pattern visit = Pattern.compile("(?<=something over in )(.*)(?= and he'd like)");
+    Matcher m = visit.matcher(responseText);
+    if (m.find()) {
+      location = m.group(0);
+    } else {
+      // He's just sitting there, waiting for you to bring his package back from <location>.
+      visit = Pattern.compile("(?<=back from )(.*)(?=\\.</td>)");
```

Comment:
You don't need a positive lookbehind here, just lose the brackets. also pull out the regexes into statics!

Reply:
See previous comment about me not being especially proficient with regex's.  Do you mean what I call parentheses when you say bracket?  Does your comment apply if  the interesting part of the response is 

`<tr><td>He's just sitting there, waiting for you to bring his package back from The Haunted Storage Room.</td></tr>`

with HTML before and after?  I had a simpler regex but I then had to find a period and throw it away and every after it.  Do you suggest making both of my pattern's static, final and then having a much larger scope?  Is compiling a pattern that expensive or are there other reasons you prefer that?

I'll gladly do what is asked but I do want to understand "why?" so there is at least a chance I will do things differently next time.  Thanks.

Reply:
if the only possible message is that one, then:
```
back from (.+?)\\.</td>
```
should do it.

We may as well compile patterns we use more than once for efficiency resasons. If they're not static / final they'll be recompiled every time they get used.

Reply:
The patterns are static final now.  The suggested changes are not working, or being made properly, so I need to try and figure out what isn't going on.

`Pattern.compile("back from (.+?)\\.</td>");`

did not work which is my understanding of the suggestion.

Reply:
I've changed it to what I intended. I suspect you didn't change `group(0)` to `group(1)`. Group 1 is the first captured group, while group 0 is the entire matched string (which doesn't include the parts from lookaheads or lookbehinds).

Reply:
Thank you.  I am somewhat chagrined that my biggest issue in simplifying the regex was using the wrong group.  Maybe I will remember that next time.  I tested this "live" before this last change and saw what I expected.  I'm going to consider this resolved.  Thanks again.